### PR TITLE
[6/6] VeNft - Add unlock, extend, redeem modals and functionality

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

Adds functionality for the Unlock, Redeem, and Extend Lock modals on user tokens.

## Screenshots or screen recordings

image

image

image

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/veNft/VeNftExtendLockModal.tsx [hunk 1]
```diff
@@ -1,30 +1,122 @@
...
+import { ThemeContext } from 'contexts/themeContext'
+import { useExtendLockTx } from 'hooks/veNft/transactor/VeNftExtendLockTx'
+import { NetworkContext } from 'contexts/networkContext'
+import { t, Trans } from '@lingui/macro'
+import { emitSuccessNotification } from 'utils/notifications'
+import { VeNftToken } from 'models/subgraph-entities/v2/venft-token'
 
-interface VeNftExtendLockModalProps {
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { useForm } from 'antd/lib/form/Form'
```
Comment (tomquirk) [inline src/components/veNft/VeNftExtendLockModal.tsx:12]: **suggestion**: We can avoid this import by using `Form.useForm` below instead.

File: src/components/veNft/VeNftExtendLockModal.tsx [hunk 2]
```diff
@@ -1,30 +1,122 @@
...
+  onCompleted,
 }: VeNftExtendLockModalProps) => {
-  const redeem = () => {
-    return
+  const { userAddress, onSelectWallet } = useContext(NetworkContext)
+  const { tokenId } = token
+  const {
+    veNft: { lockDurationOptions },
+  } = useContext(V2ProjectContext)
+  const [form] = useForm<ExtendLockFormProps>()
```
Comment (tomquirk) [inline src/components/veNft/VeNftExtendLockModal.tsx:38]:
```suggestion
  const [form] = Form.useForm<ExtendLockFormProps>()
```

File: src/components/veNft/VeNftExtendLockModal.tsx [hunk 3]
```diff
@@ -1,30 +1,122 @@
...
+      },
+    )
+
+    if (!txSuccess) {
+      setLoading(false)
+    }
   }
 
   return (
     <Modal
```
Comment (tomquirk) [inline src/components/veNft/VeNftExtendLockModal.tsx:94]: **suggestion**: we might consider using `TransactionModal` here (if it makes sense to)

File: src/components/veNft/VeNftExtendLockModal.tsx [hunk 4]
```diff
@@ -1,30 +1,122 @@
...
+      onOk={extendLock}
+      okText={`Extend Lock`}
+      confirmLoading={loading}
     >
       <h2>
         <Trans>Extend Lock</Trans>
       </h2>
+      <div style={{ color: colors.text.secondary }}>
+        <p>
+          <Trans>Set an updated duration for your staking position.</Trans>
```
Comment (tomquirk) [inline src/components/veNft/VeNftExtendLockModal.tsx:106]:
```suggestion
          <Trans>Update your staked position's lock duration.</Trans>
```

**Review by tomquirk — APPROVED** [review]
@jmilldotdev this one pretty much perfect. I had UI suggestions but I think they'll all be addressed upstream (from https://github.com/jbx-protocol/juice-interface/pull/1493)

LFG!

File: src/components/veNft/VeNftExtendLockModal.tsx (hunk 1 above)
Comment (jmilldotdev) [reply src/components/veNft/VeNftExtendLockModal.tsx:12]: May just be worth fixing this globally across the codebase. There are lots of other components using this import.

File: src/components/veNft/VeNftExtendLockModal.tsx (hunk 3 above)
Comment (jmilldotdev) [reply src/components/veNft/VeNftExtendLockModal.tsx:94]: Agreed.

File: src/components/veNft/VeNftExtendLockModal.tsx (hunk 4 above)
Comment (jmilldotdev) [reply src/components/veNft/VeNftExtendLockModal.tsx:106]: What about `Update your veNFT's lock duration.`, in the spirit of trying to pick staked or locked universally (i lean towards locked)

**Comment by jmilldotdev** [comment]
All addressed

**Pull request merged**